### PR TITLE
Add News homepage

### PR DESCRIPTION
**Pull request by papertek**
i really dont like the code but its working. maybe this needs further thought please @emmbience @Aisuruneko

### CODE REVIEW

File: src/router/index.js [hunk 1]
```diff
@@ -11,12 +12,17 @@ const router = createRouter({
...
+      component: NewsView
+    },
+	  {
       path: '/oauthComplete',
       name: 'oauthComplete',
       component: oauthComplete
-	},
-	{ // redirect everything else to article
+	  },
+	  { // redirect everything else to article
```
Comment (emmbience) [inline src/router/index.js:25]: Tabbing is misaligned here

File: src/views/news/components/BigNewsPost.vue [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            font-size: 18px;
+            max-height: 4em;
+            overflow: hidden;
+            text-overflow: ellipsis;
+            display: -webkit-box;
+            -webkit-line-clamp: 3;
+            -webkit-box-orient: vertical;
+        }
+    }
+}
```
Comment (emmbience) [inline src/views/news/components/BigNewsPost.vue:93]: I think I was the one who did this, but the styling here is *almost* the same as in `NewsPost.vue`. To make code more DRY could do some SCSS inheritance (if that's doable in Vue, idk)

File: src/views/home/components/BigPost.vue [hunk 3]
```diff
@@ -26,7 +26,8 @@ const props = defineProps({
 
 <template>
     <RouterLink :to="post.url || '/not-found'" class="w-fit">
-        <OverlapGrid class="big-post w-full md:w-80 h-full rounded-lg overlap-grid" v-if="post.meta">
+        <!-- md:w-80 -->
```
Comment (emmbience) [inline src/views/home/components/BigPost.vue:29]: What is this comment for?

File: src/views/news/components/BigNewsPost.vue (hunk 2 above)
Comment (Aisuruneko) [reply src/views/news/components/BigNewsPost.vue:93]: thinking we should do this once this is merged. we can do global scss, but i dont wanna blow up anyone else's PR

File: src/router/index.js (hunk 1 above)
Comment (Aisuruneko) [reply src/router/index.js]: fixed in [731587e](https://github.com/CamelliaCommunity/wiki-frontend/pull/16/commits/731587eb2dd5a8b8695919fc6577fe2c4546b3ed)

File: src/views/home/components/BigPost.vue (hunk 3 above)
Comment (papertek) [reply src/views/home/components/BigPost.vue:29]: just old code. removed it

File: src/views/news/components/BigNewsPost.vue (hunk 2 above)
Comment (papertek) [reply src/views/news/components/BigNewsPost.vue:93]: Definitely. in a cleanup pr maybe

Comment (Aisuruneko) [reply src/views/news/components/BigNewsPost.vue:93]: I planned on doing that once the meta tags was implemented, its just that is taking a painstakingly long time to fix things to make it ready to merge.

Comment (papertek) [reply src/views/news/components/BigNewsPost.vue:93]: 🥉

**Review by Aisuruneko — APPROVED** [review]
blows up review cutely

**Pull request merged**